pandas/core/groupby.py

```diff
@@ -794,6 +794,7 @@ def _concat_objects(self, keys, values, not_indexed_same=False):
 
             if isinstance(result, Series):
                 result = result.reindex(ax)
+                result.name = self.name
```

Comment:
@sinhrks this fixed a couple of tests in groupby below, but I don't know if we have a related issue.

related (but did not fix) #12363